### ISSUE TITLE
loginFlow/Delete appPassword not working

### ISSUE THREAD

**Issue by Gabrielhn** [nextcloud/documentation#1688]
I'm trying to call the api method to **delete the appPassword** of the client device when my users logoff from the app. Found the a little help in the docs, but still can't call it with a 200 return . Already tried with these:

1)
Method = `DELETE`.
Headers = (Basic with user and appPassword encoded) | 'OCS-APIRequest': 'true'
URL =`https://host.123/ocs/v2.php/core/KjGzyhR7B1OEEy1noUyFvvTG0ywxdTIx4p33WHMoYitCGiQO7RFkVsMdvlnsq4xL1VpbnGKA`

2)
Method = `DELETE`.
Headers = (Basic with user and appPassword encoded) | 'OCS-APIRequest': 'true'
URL =`https://host.123/ocs/v2.php/core/apppassword`

What's is the right way to do it?

**Comment by skjnldsv** [comment]
cc @ChristophWurst

**Comment by kesselb** [comment]
> but still can't call it with a 200 return

What response do you get? and what Nextcloud version do you use?

**Comment by Gabrielhn** [comment]
Hi @kesselb .
**Version**: 17.0.1

calling:
`curl -u cxd:qo3p8PZuiK5cZ1ZeU47yaV1SisTIu9TwUb43Wiy10G5286gvueyz0aISkIUIk6oA4jSZ6CXj -X DELETE -H 'OCS-APIREQUEST: true'  https://cleandrive.cleannet.com.br/ocs/v2.php/core/qo3p8PZuiK5cZ1ZeU47yaV1SisTIu9TwUb43Wiy10G5286gvueyz0aISkIUIk6oA4jSZ6CXj`

response:
`<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>404</statuscode>
  <message>Invalid query, please check the syntax. API specifications are here: http://www.freedesktop.org/wiki/Specifications/open-collaboration-services.
</message>
 </meta>
 <data/>
</ocs>`

**Comment by kesselb** [comment]
`curl -u username:app-password -X DELETE -H 'OCS-APIREQUEST: true'  http://localhost/ocs/v2.php/core/apppassword`

does not work for you? :thinking:

**Comment by Gabrielhn** [comment]
Now it worked, must send the keyword "apppassword" at the end not the actual appPassword

Thanks!

**Comment by Gabrielhn** [comment]
I'm reopening the issue because are facing another related problem. 
I'm developing a flutter app connected to a Nextcloud hosted server and at this moment trying to implement the logoff process, which includes the first part of the this issue, to make the deleteAppPassword method working.

I'll explain the steps to what I'm seeing is causing my problem.

1) When the user first log in to the app, request and authorize the appPassword 2 lines are created in the security section of the web client, one for the appPassword session and one for the device, I think.

<img width="685" alt="Captura de Tela 2019-11-20 às 18 55 15" src="https://user-images.githubusercontent.com/24768037/69281672-75255600-0bc7-11ea-8f62-4c8d7c548dae.png">

2) After this all the actions the user prompt within the app are authenticated using the appPassword.

3) When implementing the logoff process at this moment I'm calling the delete appPassword method and it's working fine, after that the sessions screen looks like this without the appPassword session, but the device still there:

<img width="719" alt="Captura de Tela 2019-11-20 às 19 00 01" src="https://user-images.githubusercontent.com/24768037/69281972-05639b00-0bc8-11ea-8c5c-2d7a2530ade0.png">

4) After the logoff process finished, I redirect the user to the login screen again, to restart and maybe login with another account, but because the actual device still listed in the security section the login screen don't prompt the user to insert another credentials just ask to authorize the device with a new appPassword and log him again.

5)I'm tested to manually remove the device on the web client before starting the login process again and the it works as expected.

**THE PROBLEM: Is there a way (with API) to actually remove the device too from that list too?**

**Comment by kesselb** [comment]
> Is there a way (with API) to actually remove the device too from that list too

I don't think so. It's a session (from the login flow) and will be removed by nextcloud.

**Comment by williamrech** [comment]
As mentioned above, I have the exact same problem, two tokens are being created. However, I have tested the same account on the NextCloud App and it created one token only, so, presumably there is something different from the way that used the API's and the documentation.

Adding to the open question, is there a way to prevent this behavior or maybe revoking two tokens instead of one?

**Comment by skjnldsv** [comment]
This is now documented. `developer_manual/client_apis/LoginFlow/index.rst` shows the correct endpoint: `DELETE /ocs/v2.php/core/apppassword` authenticated with the app password itself. The token should not be appended to the URL path.
